### DISCUSSION TITLE
Does the Main Branch Support FLUX Training?

### DISCUSSION THREAD

**Discussion by ArmadstheDoom** [kohya-ss/sd-scripts#1604] [Q&A]
So I know there's an offshoot branch/different version that some other person made, but as far as I know that one is now behind the main version? So does the main version support FLUX training? If not, how do you pull from a branch version since git pull alone isn't enough and nothing is compiled for us smooth brained non-programmers.

I can't seem to figure out if the main version has been updated in it's support; I know civitai runs off of koyha, so clearly they're using something, but I have no idea if that thing is the main branch or how to access it. 

So if it doesn't support flux in the main branch, when will it, and what commands do you need to pull the downloads from the branch version if not?

**Comment by deefster** [comment]
Your question seems to be around git usage, but I'll caution now that even after you do all this, you'll have to spend some time with the README.md file to understand the things you need to do specific to flux training to get everything working.  It's not just launch the gui and go. You need a specific pytorch version, you need to run the training via cli, etc.

Also I'm a linux guy, so you might have to translate this if you're on some inferior OS
I did the following

first clone it like usual

git clone https://github.com/kohya-ss/sd-scripts.git
cd kohya_ss/sd_scripts

now checkout the sd3 branch (the flux training code is in this branch)

git checkout sd3

I'm on an older commit, but the above is what I did.  When you open the README you'll see a bunch of comments about flux training.  You'll also see a flux_train.py file.  If you don't see these things, something with your checkout failed.

**Comment by WVVWVWVVW** [comment]
Open up cmd or whatever you use in your kohya_ss folder and type git switch -f sd3-flux.1
